File: 9_single_inheritance.cpp
#include<bits/stdc++.h>

using namespace std;
#define fast ios_base::sync_with_stdio(0); cin.tie(NULL);
#define ll long long

class A{
private:
    int value;
public:
    string name;

    void set_value(int val)
    {
        value = val;
    }

    int get_value()
    {
        return value;
    }
protected:
    int age;

};

class B:public A{
public:
    B(int val,string name,int age)
    {
        // value = val;
        set_value(val);
        this->name = name;
        this->age = age;
    }


    void display()
    {
        cout<<"Name -> "<<name<<endl;
        cout<<"Age -> "<<age<<endl;
        cout<<"Value -> "<<get_value()<<endl;
    }

};

void solve()
{
    A obj;

    obj.name = "Elias";
    // obj.age = 19;

    cout<<obj.name<<endl;
    // cout<<obj.age<<endl;

    B newObj(10,"She",90);
    // newObj.name = "Karim";
    
    cout<<newObj.name<<endl;
    newObj.display();
}

int main()
{
    fast;
    solve();
    return 0;
}

File: 9_single_inheritance.py
class A:
    __value = 0
    name = ''
    _age = 0
    
    def setter(self,val):
        self.__value = val
    
    def getter(self):
        return self.__value
    
    
class B(A):
    def __init__(self,val,name,age):
        self.name = name
        self._age = age
        self.setter(val)
        
    def display(self):
        # print(self.__value)
        print(self.getter())
        print(self.name)
        print(self._age)
    

obj = A()

# obj.name = "Elias"
# obj.__value = 90
# obj.setter(90)
# obj._age = 10


# print(A.__dict__)
# print(obj.__dict__)

newObj = B(40,'Karim',45)
newObj.display()

print(newObj.__dict__)

# obj.display()